app/pages/lab-pages-editor/helpers/cleanupTasksAndSteps.js

```diff
@@ -10,32 +10,41 @@ Clean up tasks and steps.
 
 export default function cleanupTasksAndSteps(tasks = {}, steps = []) {
   const newTasks = structuredClone(tasks);  // Copy tasks
-  let newSteps = steps.slice();  // Copy steps
+  let newSteps = structuredClone(steps);  // Copy steps. This is a deep copy, compared to steps.slice()
+
+  // Remove steps without tasks.
+  newSteps = newSteps.filter(step => step?.[1]?.taskKeys?.length > 0);
+
+  // Remove tasks not associated with any step.
+  Object.keys(newTasks).forEach(taskKey => {
+    let existsInAnyStep = false;
+    newSteps.forEach(step => {
+      existsInAnyStep = existsInAnyStep || !!step?.[1]?.taskKeys?.includes(taskKey);
+    });
+    if (!existsInAnyStep) delete newTasks[taskKey];
+  });
```

Comment:
Just linking this FEM [Issue](https://github.com/zooniverse/front-end-monorepo/issues/5807) about orphaned tasks for documentation. Your implementation here means any project using the Pages Editor solves the bug 🎉 Hopefully all FEM projects soon 🤞